aten/src/ATen/native/vulkan/VulkanAten.cpp

```diff
@@ -413,23 +425,30 @@ Tensor vulkan_clamp(
     const Tensor& self,
     const c10::optional<Scalar> min,
     const c10::optional<Scalar> max) {
-  VulkanTensor& x = vtensor_from_vulkan(self);
-  VulkanTensor output = VulkanTensor{self.sizes().vec()};
+  const auto& x = vtensor_from_vulkan(self);
+  VulkanTensor output{self.sizes().vec()};
   output.allocate_storage();
-  float minValue = min.has_value() ? min.value().to<float>()
-                                   : std::numeric_limits<float>::min();
-  float maxValue = max.has_value() ? max.value().to<float>()
-                                   : std::numeric_limits<float>::max();
-  vulkan::detail::clamp(output, x, minValue, maxValue);
+  vulkan::detail::clamp(
+      output,
+      x,
+      min ? min.value().to<float>() : -std::numeric_limits<float>::infinity(),
+      max ? max.value().to<float>() : std::numeric_limits<float>::infinity());
   return new_with_vtensor_vulkan(std::move(output), self.options());
 }
 
-Tensor& _clamp__vulkan(
+Tensor& vulkan_clamp_(
     Tensor& self,
     const c10::optional<Scalar> min,
     const c10::optional<Scalar> max) {
-  auto y = vulkan_clamp(self, min, max);
-  self.copy_(y);
+  auto& x = vtensor_from_vulkan(self);
+  VulkanTensor output{self.sizes().vec()};
+  output.allocate_storage();
+  vulkan::detail::clamp(
+      output,
+      x,
+      min ? min.value().to<float>() : -std::numeric_limits<float>::infinity(),
+      max ? max.value().to<float>() : std::numeric_limits<float>::infinity());
+  x = std::move(output);
```

Comment:
Same.  If this operation is in place, then why do we need to `allocate_storage`?

Reply:
(As in previous comment) Output VulkanTensor is used here as a temporary variable to reuse shaders with separate output, inputs; allocate_storage to remove in https://github.com/pytorch/pytorch/pull/42569 :)